docs/source/examples/batch-inference.rst

```diff
@@ -0,0 +1,315 @@
+
+Split Data into Smaller Chunks
+------------------------------
+
+[LMSys-1M dataset](https://huggingface.co/datasets/lmsys/lmsys-chat-1m) contains 1M conversations with 6 parquet files:
+
+.. code-block::
+
+    train-00000-of-00006-*.parquet
+    train-00001-of-00006-*.parquet
+    train-00002-of-00006-*.parquet
+    train-00003-of-00006-*.parquet
+    train-00004-of-00006-*.parquet
+    train-00005-of-00006-*.parquet
+
+
+For simplicity, in order to scale out the inference to more than 6 nodes, we can firstly split the entire dataset into smaller chunks.
+
+.. note::
+
+    This step is optional. We offer a R2 bucket with LMSys-1M dataset that is already divided into smaller chunks: ``r2://skypilot-lmsys-chat-1m`` (Check it in browser `here <https://pub-109f99b93eac4c22939d0ed4385f0dcd.r2.dev>`_).
+
+.. TODO: confirm r2 bucket's public access
+
+We first start a machine for development. The following command will start a small CPU machine with the current directory synced and a R2 bucket mounted, so we can directly operate on the data as if they are on local disk.
+
+.. code-block:: bash
+
+    sky launch -c dev dev.yaml --workdir .
+    # SSH into the remote machine.
+    ssh dev
+    # Running on remote machine.
+    cd sky_workdir
+
+On the remote dev machine, we download the LMSys-1M dataset and split it into smaller chunks, using a utility script:
+
+.. code-block:: bash
+
+    python utils/download_and_convert.py
+
+This script converts the dataset into 200 data chunks, each containing 5000 conversations. A metadata file is also generated with all paths to data chunks:
+
+.. code-block::
+  
+    metadata.txt
+    part_0.jsonl
+    part_1.jsonl
+    ...
+    part_199.jsonl
+
+.. note::
+
+    We use R2 bucket for its no data egress fee, so we can easily scale out the inference to multiple regions/clouds without additional costs for data reading.
+
+
+.. _develop-inference-script:
+
+Develop Inference Script
+------------------------
+
+With the data splitted into smaller chunks, we can now develop an inference script to generate predictions for each chunk.
+
+First of all, we start another dev machine with GPUs so we can develop and debug the inference script by directly logging into the machine and running the script.
+
+.. code-block:: bash
+
+    sky launch -c dev dev.yaml --gpus L4 --workdir .
+    ssh dev
+    cd sky_workdir
+
+We now develop the inference script to generate predictions for the first turn of each conversation in LMSys-1M dataset. 
+
+The following is an example script, where we aggregate multiple inputs into a single batch for better GPU utilization, and process the entire chunk of data batch by batch:
+
+.. code-block:: python
+    
+    from vllm import LLM
+    
+    BATCH_CHAR_COUNT = 2000
+    DATA_PATH = '/data/part_0.jsonl'
+    OUTPUT_PATH = '/output'
+
+    llm = LLM(model='meta-llama/Meta-Llama-3.1-7B-Instruct', tensor_parallel_size=1)
+
+    def batch_inference(llm: LLM, data_path: str):
```

Comment:
Rename? It seems to be working on 1 work item / 1 batch only, and there's no 'batching'?

Reply:
Yes, it should be renamed to "infer_one_batch()".